mlflow/projects/__init__.py

```diff
@@ -185,48 +115,103 @@ def _get_db_hostname_and_auth():
         return config.host, config.token, config.username, config.password
 
 
-def _run_databricks(uri, entry_point, version, parameters, experiment_id, cluster_spec,
-                    git_username, git_password):
-    hostname, token, username, password, = _get_db_hostname_and_auth()
+def _do_databricks_run(project_uri, command, env_vars, cluster_spec):
+    hostname, token, username, password, = _get_databricks_hostname_and_auth()
     auth = (username, password) if username is not None and password is not None else None
-    # Read cluster spec from file
-    with open(cluster_spec, 'r') as handle:
-        cluster_spec = json.load(handle)
     # Make jobs API request to launch run.
-    env_vars = {"MLFLOW_GIT_URI": uri}
-    if git_username is not None:
-        env_vars["MLFLOW_GIT_USERNAME"] = git_username
-    if git_password is not None:
-        env_vars["MLFLOW_GIT_PASSWORD"] = git_password
-    # Pass experiment ID to shell job on Databricks as an environment variable.
-    if experiment_id is not None:
-        eprint("=== Using experiment ID %s ===" % experiment_id)
-        env_vars[tracking._EXPERIMENT_ID_ENV_VAR] = experiment_id
     req_body_json = {
-        'run_name': 'MLflow Job Run for %s' % uri,
+        'run_name': 'MLflow Run for %s' % project_uri,
         'new_cluster': cluster_spec,
         'shell_command_task': {
-            'command': _get_databricks_run_cmd(uri, entry_point, version, parameters),
+            'command': command,
             "env_vars": env_vars
         },
         "libraries": [{"pypi": {"package": "mlflow==%s" % VERSION}}]
     }
-    eprint("=== Running entry point %s of project %s on Databricks. ===" % (entry_point, uri))
     run_submit_res = rest_utils.databricks_api_request(
         hostname=hostname, endpoint="jobs/runs/submit", token=token, auth=auth, method="POST",
         req_body_json=req_body_json)
-    run_id = run_submit_res["run_id"]
+    job_run_id = run_submit_res["run_id"]
     eprint("=== Launched MLflow run as Databricks job run with ID %s. Getting run status "
-           "page URL... ===" % run_id)
+           "page URL... ===" % job_run_id)
     run_info = rest_utils.databricks_api_request(
         hostname=hostname, endpoint="jobs/runs/get", token=token, auth=auth, method="GET",
-        params={"run_id": run_id})
+        params={"run_id": job_run_id})
     jobs_page_url = run_info["run_page_url"]
     eprint("=== Check the run's status at %s ===" % jobs_page_url)
+    return job_run_id
+
+
+def _create_databricks_run(experiment_id, source_name, source_version, entry_point_name):
+    # Figure out tracking URI
+    tracking_uri = tracking.get_tracking_uri()
+    if tracking._is_local_uri(tracking_uri):
```

Comment:
**nit** : let's make `tracking._is_local_uri` public since it is being used outside the package.